driver-core/src/main/java/com/datastax/driver/core/HostConnectionPool.java

```diff
@@ -337,30 +289,49 @@ private PoolingOptions options() {
       }
     }
 
-    if (connections[shardId].isEmpty()) {
-      if (host.convictionPolicy.canReconnectNow()) {
-        if (connectionsPerShard == 0) {
-          maybeSpawnNewConnection(shardId);
-        } else if (scheduledForCreation[shardId].compareAndSet(0, connectionsPerShard)) {
-          for (int i = 0; i < connectionsPerShard; i++) {
-            // We don't respect MAX_SIMULTANEOUS_CREATION here because it's  only to
-            // protect against creating connection in excess of core too quickly
-            manager.blockingExecutor().submit(new ConnectionTask(shardId));
-          }
-        }
-        return enqueue(timeout, unit, maxQueueSize, shardId);
-      }
+    int index = 0;
+    int shId = shardId;
+    do {
+    	synchronized(connections[shId]) {
+    		if ((connections[shId].isEmpty()) && (index < connections.length)) {
```

Comment:
Nit: Unnecessary brackets

Reply:
In Java, braces should not be ommitted in one liners. Check this: https://dzone.com/articles/omitting-braces-not-just-a-mat; Also a good reference here: https://www.amazon.com/Effective-Java-Joshua-Bloch/dp/0134685997

Reply:
I didn't suggest doing:
```
if ((connections[shId].isEmpty()) && (index < connections.length))
  shId = index++;
```
I suggested doing:
```
if (connections[shId].isEmpty() && index < connections.length) {
  shId = index++;
}
```
No need to put an expression in a brackets when it's as simple as connections[shId].isEmpty() or index < connections.length